source/common/http/conn_manager_impl.cc

```diff
@@ -12,6 +12,7 @@
 #include "envoy/event/dispatcher.h"
 #include "envoy/network/drain_decision.h"
 #include "envoy/router/router.h"
+#include "envoy/server/admin.h"
```

Comment:
drive by follow up comment: I'm not crazy about including admin here and then special casing it with dynamic casts. Is it not possible for this logical to be external to the construction of the HCM and then specified correctly either by admin or by the normal filter?

Reply:
we've discussed it offline, I will make a cleanup/refactor PR to move the dynamic_cast outside of conn_manager_impl.